clients/java/src/main/java/marquez/client/MarquezUrl.java

```diff
@@ -45,14 +46,14 @@ static MarquezUrl create(URL url) {
   }
 
   @VisibleForTesting
-  URL from(String path) {
+  URL from(List<String> path) {
     return from(path, ImmutableMap.of());
   }
 
   @VisibleForTesting
-  URL from(String path, @Nullable Map<String, Object> queryParams) {
+  URL from(List<String> path, @Nullable Map<String, Object> queryParams) {
     try {
-      final URIBuilder builder = new URIBuilder(baseUrl.toURI()).setPath(baseUrl.getPath() + path);
+      final URIBuilder builder = new URIBuilder(baseUrl.toURI()).setPathSegments(path);
```

Comment:
we can with 
`final URIBuilder builder = new URIBuilder(URI.create(baseUrl.toURI()+path));`
if it would be encoded previously in MarquezPathV1